src/ui/components/BuildProgressBar/index.tsx

```diff
@@ -60,6 +60,21 @@ const BuildProgressBar: FunctionComponent<BuildProgressBarProps> = memo(
             default:
           }
           break;
+        case BuildJobType.ForceFlash:
+          switch (notification.step) {
+            case BuildFirmwareStep.VERIFYING_BUILD_SYSTEM:
+              return 5;
+            case BuildFirmwareStep.DOWNLOADING_FIRMWARE:
+              return 15;
+            case BuildFirmwareStep.BUILDING_USER_DEFINES:
+              return 28;
+            case BuildFirmwareStep.BUILDING_FIRMWARE:
+              return 56;
+            case BuildFirmwareStep.FLASHING_FIRMWARE:
+              return 89;
+            default:
+          }
+          break;
         default:
           throw new Error(`unhandled job type: ${jobType}`);
```

Comment:
I don't think CheckTarget will be initiated on its own, but part of another flow like "Build & Flash" to check the target before any action is performed, so it doesn't need to have its own progress bar logic, although that doesn't currently work either, see my comment on the BuildJobType.  If you think it would be useful to be a standalone action a user can take, I can add logic for it here.